### ISSUE TITLE
ShareRenamer & NextCloud 27

### ISSUE THREAD

**Issue by mateitudor** [JonathanTreffler/sharerenamer#250]
Hello, are there any plans to make your awesome plugin available in NextCloud 27?
Thanks in advance.

**Comment by EricThi** [comment]
Thank you for that !

Just report, i test on preprod today V27 and get 10msg error per seconde : 
`[sharerenamer] Fatal: OCP\AutoloadNotAllowedException: Autoload path not allowed: /var/cloud/apps/sharerenamer/appinfo/application.php at <<closure>>

0. /var/cloud/lib/autoloader.php line 166
   OC\Autoloader->isValidPath()
1. <<closure>>
   OC\Autoloader->load()
2. /var/cloud/lib/private/AppFramework/Bootstrap/Coordinator.php line 217
   class_exists()
3. /var/cloud/lib/private/App/AppManager.php line 400
   OC\AppFramework\Bootstrap\Coordinator->isBootable()
4. /var/cloud/lib/private/App/AppManager.php line 216
   OC\App\AppManager->loadApp()
5. /var/cloud/lib/private/legacy/OC_App.php line 126
   OC\App\AppManager->loadApps()
6. /var/cloud/lib/base.php line 1044
   OC_App::loadApps()
7. /var/cloud/index.php line 36
   OC::handleRequest()

POST /settings/apps/disable`

**Comment by saschabrockel** [comment]
Same here:

`{"reqId":"CqhKOdjVjYSjuKHidNsD","level":4,"time":"2023-06-16T02:56:45+00:00","remoteAddr":"111.27.0.24","user":"9a84365c-8fa9-103d-820a-6d4e0dd12d6c","app":"sharerenamer","method":"POST","url":"/settings/apps/disable","message":"Error during app loading: Autoload path not allowed: /var/www/html/custom_apps/sharerenamer/appinfo/application.php","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36","version":"27.0.0.8","exception":{"Exception":"OCP\\AutoloadNotAllowedException","Message":"Autoload path not allowed: /var/www/html/custom_apps/sharerenamer/appinfo/application.php","Code":0,"Trace":[{"file":"/var/www/html/lib/autoloader.php","line":166,"function":"isValidPath","class":"OC\\Autoloader","type":"->","args":["/var/www/html/custom_apps/sharerenamer/appinfo/application.php"]},{"function":"load","class":"OC\\Autoloader","type":"->","args":["OCA\\ShareRenamer\\AppInfo\\Application"]},{"file":"/var/www/html/lib/private/AppFramework/Bootstrap/Coordinator.php","line":217,"function":"class_exists","args":["OCA\\ShareRenamer\\AppInfo\\Application"]},{"file":"/var/www/html/lib/private/App/AppManager.php","line":400,"function":"isBootable","class":"OC\\AppFramework\\Bootstrap\\Coordinator","type":"->","args":["sharerenamer"]},{"file":"/var/www/html/lib/private/App/AppManager.php","line":216,"function":"loadApp","class":"OC\\App\\AppManager","type":"->","args":["sharerenamer"]},{"file":"/var/www/html/lib/private/legacy/OC_App.php","line":126,"function":"loadApps","class":"OC\\App\\AppManager","type":"->","args":[[]]},{"file":"/var/www/html/lib/base.php","line":1044,"function":"loadApps","class":"OC_App","type":"::","args":[]},{"file":"/var/www/html/index.php","line":36,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"/var/www/html/lib/autoloader.php","Line":141,"message":"Error during app loading: Autoload path not allowed: /var/www/html/custom_apps/sharerenamer/appinfo/application.php","exception":[],"CustomMessage":"Error during app loading: Autoload path not allowed: /var/www/html/custom_apps/sharerenamer/appinfo/application.php"},"id":"648bcf7503492"}`

**Comment by JonathanTreffler** [comment]
I am quite busy, but I will try to get around to creating an update for Nextcloud 27 in the next days.

**Comment by cypress-hill-car** [comment]
Thank you very much @JonathanTreffler.

**Comment by Mannshoch** [comment]
Any Update here?

**Comment by bcutter** [comment]
Plenty of `OCP\AutoloadNotAllowedException: Autoload path not allowed: /var/www/nextcloud/apps/sharerenamer/appinfo/application.php` every second, `nextcloud.log` is completely spammed once upgraded to NC 27.

Please update this app to be compatible with NC v27.

**Comment by Pinkbyte** [comment]
As a workaround you can rename `apps/sharerenamer/appinfo/application.php` to`apps/sharerenamer/appinfo/app.php`, but according to Nextcloud sources(`/lib/private/App/AppManager.php`) this is deprecated.
```
$this->logger->debug('/appinfo/app.php is deprecated, use \OCP\AppFramework\Bootstrap\IBootstrap on the application class instead.'
```

**Comment by bcutter** [comment]
OK so there needs to be some work done in the code to make it compatible again with NC v27. NC 27 out already for more than one month so I thought this app was NC27 ready already.

Sit, wait, let devs do their magic. Thank your for this great app by the way 👍

**Comment by SokoFromNZ** [comment]
I am (im)patiently waiting for NC v27 support here :)

**Comment by jjroysdon** [comment]
All ya'all wanting support, have you hit that Sponsor link?  Otherwise, enjoy that NC26 is still supported with the v26.0.5 release.

**Comment by xidoc** [comment]
news ?

**Comment by Apflkuacha** [comment]
Finally I ended up in using Configurable Share Links, which is providing the same functionality and it supports NC27.
It could be found in the [Nextcloud App Store](https://apps.nextcloud.com/apps/cfg_share_links) or here: [https://github.com/jimmyl0l3c/cfg_share_links](https://github.com/jimmyl0l3c/cfg_share_links)

**Comment by bcutter** [comment]
Even I am totally happy with the features of ShareRenamer, that sounds interesting. Supports NC 27 for quite a while according to releases.

![image](https://github.com/JonathanTreffler/sharerenamer/assets/13799156/b0c2dcc9-28de-41e8-91d5-4945996fa602)

**Comment by dawidmachon** [comment]
Also waiting to release

**Comment by Lambazsq** [comment]
> Finally I ended up in using Configurable Share Links, which is providing the same functionality and it supports NC27. It could be found in the [Nextcloud App Store](https://apps.nextcloud.com/apps/cfg_share_links) or here: https://github.com/jimmyl0l3c/cfg_share_links

Hello, 
Is it possible to transfer/retrieve all the links created with ShareRenamer to Configurable Share Links?

Thank you

**Comment by Apflkuacha** [comment]
> > Finally I ended up in using Configurable Share Links, which is providing the same functionality and it supports NC27. It could be found in the [Nextcloud App Store](https://apps.nextcloud.com/apps/cfg_share_links) or here: https://github.com/jimmyl0l3c/cfg_share_links
> 
> Hello, Is it possible to transfer/retrieve all the links created with ShareRenamer to Configurable Share Links?
> 
> Thank you

Yeah, the available Shares could still be shown and modified. So no problem to switch to Configurable Share Links. In the beginning I had installed both, because of the known issue with NC27 now only one is left

**Comment by Lambazsq** [comment]
Thanks for your reply. In the meantime, I gave it a go with a 'let's see what happens' approach, and it worked smoothly. It's noteworthy that the share links created with ShareRenamer remain intact.

**Comment by ohader** [comment]
> I am quite busy, but I will try to get around to creating an update for Nextcloud 27 in the next few days.

Hey @JonathanTreffler, I hope you're doing fine. Do you think you have time to tackle the v27 compatibility release during the next few weeks? Thanks in advance!

**Comment by cypress-hill-car** [comment]
Amazing! I've just updated to nc28.Cheers.
-------- Ursprüngliche Nachricht --------Von: Oliver Hader ***@***.***> Datum: 16.01.24  17:34  (GMT+01:00) An: JonathanTreffler/sharerenamer ***@***.***> Cc: cypress-hill-car ***@***.***>, Comment ***@***.***> Betreff: Re: [JonathanTreffler/sharerenamer] ShareRenamer & NextCloud 27 (Issue #250) 

I am quite busy, but I will try to get around to creating an update for Nextcloud 27 in the next few days.

Hey @JonathanTreffler, I hope you're doing fine. Do you think you have time to tackle the v27 compatibility release during the next few weeks? Thanks in advance!

**Comment by mighty-services** [comment]
Hy there. I just found your post about this issue since I´m having the same error messages. 

Any updates on the progress so far?

I´m willing to implement any beta/test versions :)

**Comment by Apflkuacha** [comment]
There has been no update so long - finally I ended up in using Configurable Share Links, which is providing the same functionality and it supports NC27 and also works on NC28 although not official released for it yet.
It could be found in the [Nextcloud App Store](https://apps.nextcloud.com/apps/cfg_share_links) or here: https://github.com/jimmyl0l3c/cfg_share_links

**Comment by mighty-services** [comment]
Thanks [Apflkuacha](https://github.com/Apflkuacha). That's now also my solution :)

**Comment by JonathanTreffler** [comment]
I just released [v3.3.0](https://github.com/JonathanTreffler/sharerenamer/releases/tag/v3.3.0) of this app, that adds support for Nextcloud 29 :partying_face: 

If you install it manually and override the version check in the info.xml it is likely to work on versions 26 through 28 also, but I will not go through the trouble of checking all these old version for compatibility.

It took very little work to get this app compatible with modern versions of nextcloud, just the bootstrap logic, but it has sadly taken this long for me to get around to it. There are always just more nextcloud apps to update, than I have the time for :/

This app in general needs a lot more work, but I cannot promise I will find the time for that soon, but I will try. At least it works now like it used to work on the old versions.
